core/core.go

```diff
@@ -194,14 +210,20 @@ func (app *App) Start() error {
 	// returns any fatal errors. As it currently stands, if one of these watchers
 	// experiences a fatal error or crashes, it is difficult for us to tear down
 	// correctly.
-	if err := app.blockWatcher.StartPolling(); err != nil {
-		return err
-	}
-	log.Info("started block watcher")
 	if err := app.orderWatcher.Start(); err != nil {
 		return err
 	}
 	log.Info("started order watcher")
+	if err := app.blockWatcher.Start(); err != nil {
+		return err
+	}
+	log.Info("started block watcher")
```

Comment:
We actually want to start `BlockWatcher` after `OrderWatcher` so that `OrderWatcher` is already setup to receive the block events when `BlockWatcher` starts.